clang/lib/Driver/Driver.cpp

```diff
@@ -1377,6 +1377,22 @@ Compilation *Driver::BuildCompilation(ArrayRef<const char *> ArgList) {
   // Populate the tool chains for the offloading devices, if any.
   CreateOffloadingDeviceToolChains(*C, Inputs);
 
+  // Determine FPGA emulation status.
+  if (C->hasOffloadToolChain<Action::OFK_SYCL>()) {
+    auto SYCLTCRange = C->getOffloadToolChains<Action::OFK_SYCL>();
+    ArgStringList TargetArgs;
+    const ToolChain *TC = SYCLTCRange.first->second;
+    const toolchains::SYCLToolChain *SYCLTC =
+                static_cast<const toolchains::SYCLToolChain *>(TC);
```

Comment:
This bit seems to be annoying clang-format:
```suggestion
    const toolchains::SYCLToolChain *SYCLTC =
           static_cast<const toolchains::SYCLToolChain *>(TC);
```